scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/SignatureAlgorithmsExtension.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Softech and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Softech - initial creation
+ ******************************************************************************/
+package org.eclipse.californium.scandium.dtls;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.californium.elements.util.DatagramReader;
+import org.eclipse.californium.elements.util.DatagramWriter;
+import org.eclipse.californium.elements.util.StringUtil;
+
+public class SignatureAlgorithmsExtension extends HelloExtension {
+	    // DTLS-specific constants ////////////////////////////////////////
+
+		private static final int LIST_LENGTH_BITS = 16;
+
+		private static final int SIGNATURE_ALGORITHM_BITS = 16;
+		
+		private static final int  SIGNATURE_BITS = 8;
+		
+		private static final int HASH_BITS = 8;
+		
+		// Members ////////////////////////////////////////////////////////
+		
+		/** The list holding the supported signature and hash algorithms */
+		
+		private final List<SignatureAndHashAlgorithm> signatureAndHashAlgorithms;
+
+		
+		// Constructor ////////////////////////////////////////////////////
+
+		/**
+		 * Creates an instance using the signature algorithms and the hash algorithms codes
+		 * 
+		 * @param signatureAndHashAlgorithms
+		 */
+		public SignatureAlgorithmsExtension(SignatureAndHashAlgorithm[] signatureAndHashAlgorithms) {
```

Comment:
Even if some other extension-implementations still use a array (they require a cleanup :-) ), in this case using a List makes it easier.

Reply:
Yes, it makes it easier, I thought it was a convention to do. :) Refactored.